File: ios/dataProvider/contracts/SettingsContract.m
//
//  SettingsContract.m
//  StorjMobile
//
//  Created by Barterio on 5/31/18.
//  Copyright © 2018 Storj. All rights reserved.
//

#import "SettingsContract.h"

@implementation SettingsContract

static NSString *const _TABLE_NAME = @"settingsTable";
static NSString *const _ID = @"_id";
static NSString *const _SETTINGS_ID = @"settingsId";
static NSString *const _FIRST_SIGN_IN = @"isFirstSignIn";
static NSString *const _SYNC_STATUS = @"syncStatus";
static NSString *const _SYNC_SETTINGS = @"syncSettings";
static NSString *const _LAST_SYNC = @"lastSync";

+(NSString *) TABLE_NAME
{
  return _TABLE_NAME;
}

+(NSString *) ID
{
  return _ID;
}

+(NSString *) SETTINGS_ID
{
  return _SETTINGS_ID;
}

+(NSString *) FIRST_SIGN_IN
{
  return _FIRST_SIGN_IN;
}

+(NSString *) SYNC_STATUS
{
  return _SYNC_STATUS;
}

+(NSString *) SYNC_SETTINGS
{
  return _SYNC_SETTINGS;
}

+(NSString *) LAST_SYNC
{
  return _LAST_SYNC;
}

@end

File: android/app/src/main/java/io/storj/mobile/storjlibmodule/dataprovider/contracts/SettingsContract.java
package io.storj.mobile.storjlibmodule.dataprovider.contracts;

import android.provider.BaseColumns;

/**
 * Created by Yaroslav-Note on 3/19/2018.
 */

public final class SettingsContract implements BaseColumns {

    private SettingsContract() {}

    public final static String TABLE_NAME = "settingsTable";

    public final static String _SETTINGS_ID = "settingsId";
    public final static String _FIRST_SIGN_IN = "isFirstSignIn";
    public final static String _SYNC_STATUS = "syncStatus";
    public final static String _SYNC_SETTINGS = "syncSettings";
    public final static String _LAST_SYNC = "lastSync";

    public static String createTable() {
        return String.format(
                "create table if not exists %s (" +
                        "%s TEXT primary key not null, " +
                        "%s NUMBER DEFAULT 1, " +
                        "%s NUMBER DEFAULT 0, " +
                        "%s NUMBER DEFAULT 0, " +
                        "%s TIMESTAMP)",
                TABLE_NAME, _ID, _FIRST_SIGN_IN, _SYNC_STATUS, _SYNC_SETTINGS, _LAST_SYNC);
    }
}
